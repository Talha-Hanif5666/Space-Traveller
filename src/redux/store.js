/* eslint-disable import/no-extraneous-dependencies */
import thunk from 'redux-thunk';
import { legacy_createStore as createStore, combineReducers, applyMiddleware } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missions from './missions/missionsSlice';

const reducers = combineReducers({
  rockets: rocketsReducer,
  missionsReducer: missions,
});

const store = createStore(reducers, applyMiddleware(thunk));

export default store;
