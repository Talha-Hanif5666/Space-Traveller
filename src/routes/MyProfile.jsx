import Table from 'react-bootstrap/Table';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const mission = useSelector((state) => state.missions);
  const display = mission.missions.filter((mission) => mission.reserved === true);

  return (
    <main id="profileSpace">
      <aside className="profmission">
        <h2>My Missions</h2>
        <Table bordered hover>
          <tbody>
            {display.map((mission) => (
              <tr key={mission.id}>
                <td style={{ fontWeight: 'bold' }}>{mission.name}</td>
              </tr>
            ))}
          </tbody>
        </Table>
      </aside>
    </main>
  );
};

export default MyProfile;
