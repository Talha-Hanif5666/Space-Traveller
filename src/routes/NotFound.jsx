import React from 'react';

function NotFound() {
  return (
    <div>
      <h1>Oh!</h1>
      <h3>404</h3>
      <p>Click on any of the links above</p>
    </div>
  );
}

export default NotFound;
