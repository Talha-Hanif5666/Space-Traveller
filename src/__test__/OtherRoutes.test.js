import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import MyProfile from '../routes/MyProfile';
import NotFound from '../routes/NotFound';

describe('MyProfile test', () => {
  const mockedRockets = [
    {
      id: 'rocket1',
      name: 'Rocket 1',
      desc: 'A rocket description',
      img: 'rocket1.jpg',
      booked: true,
    },
    {
      id: 'rocket2',
      name: 'Rocket 2',
      desc: 'A rocket description',
      img: 'rocket2.jpg',
      booked: true,
    },
  ];
  test('to render all rockets and missions booked: ', () => {
    const myProfile = render(
      <Provider store={store}>
        <BrowserRouter>
          <MyProfile myRockets={mockedRockets} />
        </BrowserRouter>
      </Provider>,
    );
    expect(myProfile).toMatchSnapshot();
  });
});

describe('NotFound test', () => {
  it('renders NotFound route', () => {
    render(<NotFound />);
    expect(screen).toMatchSnapshot();
  });
});
