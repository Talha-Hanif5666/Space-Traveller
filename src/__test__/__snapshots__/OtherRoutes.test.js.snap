// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile test to render all rockets and missions booked:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        id="profileSpace"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="container row col-6"
              >
                <h2>
                  My Missions
                </h2>
                <ul
                  class="list-group"
                >
                  <li
                    class="list-group-item"
                    style="font-weight: bold;"
                  >
                    No Missions Joined
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="col"
            >
              <div>
                <h2>
                  Rockets booked
                </h2>
                <div
                  class="list-group"
                >
                  <div
                    class="fw-bold list-group-item"
                  >
                    Rocket 1
                  </div>
                  <div
                    class="fw-bold list-group-item"
                  >
                    Rocket 2
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      id="profileSpace"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="container row col-6"
            >
              <h2>
                My Missions
              </h2>
              <ul
                class="list-group"
              >
                <li
                  class="list-group-item"
                  style="font-weight: bold;"
                >
                  No Missions Joined
                </li>
              </ul>
            </div>
          </div>
          <div
            class="col"
          >
            <div>
              <h2>
                Rockets booked
              </h2>
              <div
                class="list-group"
              >
                <div
                  class="fw-bold list-group-item"
                >
                  Rocket 1
                </div>
                <div
                  class="fw-bold list-group-item"
                >
                  Rocket 2
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NotFound test renders NotFound route 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;
